rasterio/crs.pyx

```diff
@@ -473,6 +473,8 @@ cdef class CRS:
             return "OGC", matches["OGC"][0]
         elif "ESRI" in matches:
             return "ESRI", matches["ESRI"][0]
+        elif "IAU_2015" in matches:
```

Comment:
it looks like the version for the IAU authority is incorporated into the name? I had forgotten this was the case and I've seen it before in other places, but I wonder if it makes sense to do something more complicated here like a `any(['IAU' in _ for _ in matches.keys()])` here for the check and to get the appropriate match. I guess a bit of a question for @sgillies on what feels right to do?

Reply:
actually looking again at what this function does, why does rasterio attempt to check the authority at all?

Why not do something more compact like:
```python
matches = self._matches(confidence_threshold=confidence_threshold)
if len(matches) < 1:
    return None
authority, code = next(iter(matches.items()))
return authority, code[0]
```

if walrus is supported in cython:
```python
if not (matches := self._matches(confidence_threshold=confidence_threshold)):
    return None
authority, code = next(iter(matches.items()))
return authority, code[0]
```

Reply:
oh okay so I am reading the comment in the code right above more clearly now and the desire was that matches from other authorities could be found but prior to 1.2.7 EPSG was the only one considered, so the if statements were aimed to return EPSG first even if other codes had better matching scores. 

On one hand this is probably fine but requires knowledge of all the authorities ahead of time and updating this function any time a new authority is desired to be supported. Not that new authorities get added to often but that is a bit annoying to have to deal with. 

Alternatively, as there have been a few major version changes, it could make sense to change the behavior and now accept that whatever authority returns the highest score is probably the correct authority, and if EPSG is desired users can just use `.to_epsg` or call the private `_matches` function themselves. We could also make a public `authority_matches` function to empower users this way. 

I just started looking at the OGR api now to see if there is an equivalent to `pyproj.database.get_authorities`, but haven't found it yet, but also think that a list of authority priorites would need to then be maintained in rasterio like the if statements so that doesn't really change anything for the better.

Thoughts? @vincentsarago @sgillies @snowman2 ?

Reply:
> I just started looking at the OGR api now to see if there is an equivalent to `pyproj.database.get_authorities`, but haven't found it yet,

==> https://github.com/OSGeo/gdal/pull/11130

Reply:
Looks like I am late to the conversation here, but thought this information could be helpful with the design.

For pyproj, [to_authority](https://pyproj4.github.io/pyproj/stable/api/crs/crs.html#pyproj.crs.CRS.to_authority) has an optional `auth_name` argument to only return results from a specific authority. That could potentially be used here to allow users to select the authority they want it they prefer a specific authority. If it isn't provided, the best match is returned regardless of the authority.

For this function, it appears to not return a match at all if an authority that is not preferred is identified (I may be missing something because I am on my phone). If what I am seeing is the case, is there a reason not to return a match if it isn't in the prioritized authority list?